### PR TITLE
Fix osx cmake

### PR DESCRIPTION
**Pull request by guysherman**
Refers to issue #27 .

This is a replacement for a previous PR.

Further tidied CMake files as per Han Zhu's requests  …
Managed to resolve the CMAKE issue around finding OpenSSL, so now
it uses the same method as Casablanca, and doesn't require homebrew
users to specify OPENSSL_ROOT_DIR. The readme explains what to set it
to for non-homebrew users (eg Fink, or Macports)

Tidied up the process for including the -L for gettext, which is
required by libintl, which is part of the dependency chain for
libxml++, but isn't handled very well by pkg-config for some reason. 

@hanzhumsft this is were the need for gettext comes in. The dependency graph is something like libxml++ -> glibmm -> gobject -> glib-2.0 -> gettext (which is libintl and friends)

Again we assume the path for homebrew, (raising messages if we can't)
find it, and the readme describes how non-homebrew users can point
to their installation if needed.

in src/CMakeLists.txt, put an APPLE conditional around adding the warnings
variable to the CXX flags

Tidided up the readme section for OSX so that it drops some of the
linux bits that were copy pasted. Made clearer instructions for homebrew
users vs non-homebrew users. Removed some of the brew packages that the
user needs to install as they will be installed as dependencies if needed.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@guysherman__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: .gitignore [hunk 1]
```diff
@@ -182,3 +182,9 @@ $RECYCLE.BIN/
 
 # build folders under Linux
 Microsoft.WindowsAzure.Storage/build.*/
+
+# ==========================
+# OSX detritus
+# ==========================
+
+*.DS_Store
```
Comment (hanzhumsft) [inline .gitignore]: minor: just `.DS_Store`

File: README.md [hunk 2]
```diff
@@ -133,6 +133,67 @@ cd Binaries
...
+- `<path to openssl>` to your local openssl, it is recommended not to use the version that comes with OSX, rather use one from Homebrew or the like. This should be the path that contains the `lib` and `include` directories
+- `<path to gettext lib dir>` is the directory which contains `libintl.dylib`
+
+For example you might use:
+```bash
+CASABLANCA_DIR=~/Github/Casablanca/cpprestsdk cmake .. -DCMAKE_BUILD_TYPE=Release -DOPENSSL_ROOT_DIR=/usr/local/opt/openssl -DGETTEXT_LIB_DIR=/usr/local/opt/gettext/lib
+```
+The library is generated under `azure-storage-cpp/Microsoft.WindowsAzure.Storage/build.release/Binaries/`.
+
+*As yet the unit tests don't build, because cmake won't find UnitTest++ even though it is installed via Homebrew*
```
Comment (hanzhumsft) [inline README.md]: I tried `brew install unittest-cpp`, and UnitTest++ can be found.

File: .gitignore (hunk 1 above)
Comment (guysherman) [inline .gitignore]: In githubs template for OSX they have it as *.DS_Store. I'm pretty sure .DS_Store without the star didn't work.

File: README.md [hunk 3]
```diff
@@ -133,6 +133,67 @@ cd Binaries
...
+
+Once this is complete, then:
+
+- Clone the project using git:
+```bash
+git clone https://github.com/Azure/azure-storage-cpp.git
+```
+The project is cloned to a folder called `azure-storage-cpp`. Always use the master branch, which contains the latest release.
+
+**Some notes about building**:
```
Comment (hanzhumsft) [inline README.md]: How about use `brew link --force openssl`?  This command forcefully link all openssl libraries and headers to `/usr/local/`.  Any concern on this command?

File: README.md (hunk 2 above)
Comment (guysherman) [inline README.md]: Ah, that might be a result of the fix you made to the CMake module for that. Let me try again.

File: Microsoft.WindowsAzure.Storage/CMakeLists.txt [hunk 4]
```diff
@@ -15,7 +15,33 @@ option(BUILD_SAMPLES "Build sample codes" OFF)
 if(UNIX)
   find_package(Boost REQUIRED COMPONENTS log log_setup random system thread locale regex filesystem chrono date_time)
   find_package(Threads REQUIRED)
-  find_package(OpenSSL REQUIRED)
+  if(APPLE AND NOT OPENSSL_ROOT_DIR)
+    # Prefer a homebrew version of OpenSSL over the one in /usr/lib
```
Comment (hanzhumsft) [inline Microsoft.WindowsAzure.Storage/CMakeLists.txt]: How about use `brew link --force openssl`?  This command forcefully link all openssl libraries and headers to `/usr/local/`.  Then we don't have to specify `OPENSSL_ROOT_DIR` at all.  And `brew link --force gettext` can do the same for `GETTEXT_LIB_DIR`.  Any concern on using this command `brew link --force`?

Comment (guysherman) [inline Microsoft.WindowsAzure.Storage/CMakeLists.txt]: I tend to avoid `brew link --force` I've broken my system in the past with it. Even homebrew recommend against it when you install keg-only packages.

File: Microsoft.WindowsAzure.Storage/src/CMakeLists.txt [hunk 5]
```diff
@@ -54,6 +55,11 @@ endif()
 if ("${CMAKE_BUILD_TYPE}" MATCHES "Debug")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fprofile-arcs -ftest-coverage")
 endif()
+if (APPLE)
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${WARNINGS}")
+else()
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${WARNINGS}")
+endif()
```
Comment (hanzhumsft) [inline Microsoft.WindowsAzure.Storage/src/CMakeLists.txt]: `if` and `else` is same?  I think we should only have the `if` part.

File: Microsoft.WindowsAzure.Storage/CMakeLists.txt (hunk 4 above)
Comment (guysherman) [inline Microsoft.WindowsAzure.Storage/CMakeLists.txt]: Also, there's precedent for doing it this way... I borrowed most of this from the cpprestsdk CMake.

File: Microsoft.WindowsAzure.Storage/src/CMakeLists.txt (hunk 5 above)
Comment (guysherman) [inline Microsoft.WindowsAzure.Storage/src/CMakeLists.txt]: Ah, no I forgot to remove ${WARNINGS} from the else part.

**Comment by guysherman** [comment]
@hanzhumsft I have updated the PR with all your feedback except for the `brew link --force` instructions. I really don't think this project should require the developer to do something that homebrew recommend against.

File: Microsoft.WindowsAzure.Storage/CMakeLists.txt (hunk 4 above)
Comment (hanzhumsft) [inline Microsoft.WindowsAzure.Storage/CMakeLists.txt]: Make sense.  Let's not ask developers do forceful links.

**Comment by hanzhumsft** [comment]
Thanks @guysherman .  The change looks good to me.

I agree we should not ask developers do dangerous things to there system. So `brew link --force` is not recommended.
